plugins/modules/cm_kerberos.py

```diff
@@ -0,0 +1,487 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2025 Cloudera, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ansible_collections.cloudera.cluster.plugins.module_utils.cm_utils import (
+    ClouderaManagerModule,
+    ClouderaManagerMutableModule,
+    resolve_parameter_updates,
+)
+from cm_client.rest import ApiException
+from cm_client import (
+    ClouderaManagerResourceApi,
+    ApiConfigList,
+    ApiConfig,
+)
+import re
+
+DOCUMENTATION = r"""
+module: cm_kerberos
+short_description: Manage and configure Kerberos Authentication for CDP
+description:
+  - Manages Kerberos authentication and configuration in Cloudera Manager.
+  - Imports the KDC Account Manager credentials needed by Cloudera Manager to create kerberos principals.
+author:
+  - "Jim Enright (@jimright)"
+requirements:
+  - cm_client
+options:
+  state:
+    description:
+      - The declarative state of Kerberos configuration.
+    type: str
+    required: false
+    default: present
+    choices:
+      - present
+      - absent
+  krb_enc_types:
+    description:
+      - Kerberos Encryption Types supported by the KDC to set in Cloudera Manager configuration.
+    type: list
+    elements: str
+    required: false
+  security_realm:
+    description:
+      - Kerberos Security Realm to set in Cloudera Manager configuration
+      - Changing this setting would clear up all existing credentials and keytabs from Cloudera Manager.
```

Comment:
Do you mean setting this variable removes all existing credentials and keytabs if change is required?

Reply:
Yes, I've reworded this description to (hopefully!) make this clearer